src/client/components/pages/parts/pager.js

```diff
@@ -0,0 +1,134 @@
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+import * as bootstrap from 'react-bootstrap';
+import PropTypes from 'prop-types';
+import React from 'react';
+import request from 'superagent-bluebird-promise';
+
+
+const {Pager, Button, ButtonGroup, DropdownButton, MenuItem} = bootstrap;
+
+class PagerElement extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			from: this.props.from,
+			nextEnabled: this.props.results.length === this.props.size,
+			results: this.props.results,
+			size: this.props.size
+		};
+		this.handleClickPrevious = this.handleClickPrevious.bind(this);
+		this.handleClickNext = this.handleClickNext.bind(this);
+		this.handleResultsPerPageChange = this.handleResultsPerPageChange.bind(this);
+		this.triggerSearch = this.triggerSearch.bind(this);
+	}
+
+	triggerSearch() {
+		const pagination = `&size=${this.state.size}&from=${this.state.from}`;
+		const collectionString = this.state.collection ? `&collection=${this.state.collection}` : '';
+
+		request.get(`${this.props.paginationUrl}${this.props.query}${collectionString}${pagination}`)
+			.then((res) => JSON.parse(res.text))
+			.then((data) => {
+				this.setState(prevState => ({
+					nextEnabled: data.length >= prevState.size,
+					results: data
+				}));
+				this.props.changeDataInTable(data);
+			});
+	}
+
+	handleClickPrevious() {
+		this.setState(prevState => ({from: Math.max(prevState.from - prevState.size, 0)}), this.triggerSearch);
+	}
+
+	handleClickNext() {
+		this.setState(prevState => ({from: prevState.from + prevState.size}), this.triggerSearch);
+	}
+
+	handleResultsPerPageChange(value) {
+		this.setState({size: parseInt(value, 10)}, this.triggerSearch);
+	}
+
+	render() {
+		return (
+			<div id="PagerElement">
+				{
+					this.state.results && this.state.results.length ?
+						<div>
+							<hr className="thin"/>
+							<Pager>
+								<Pager.Item
+									previous disabled={this.state.from <= 0}
+									href="#" onClick={this.handleClickPrevious}
+								>
+									&larr; Previous Page
+								</Pager.Item>
+								<ButtonGroup>
+									<Button disabled>Results {this.state.from + 1} —
+										{this.state.results.length < this.state.size ?
+											this.state.from + this.state.results.length :
+											this.state.from + this.state.size
+										}
+									</Button>
+									<DropdownButton
+										dropup bsStyle="info" id="bg-nested-dropdown"
+										title={`${this.state.size} per page`}
+										onSelect={this.handleResultsPerPageChange}
+									>
+										<MenuItem eventKey="10">10 per page</MenuItem>
+										<MenuItem eventKey="20">20 per page</MenuItem>
+										<MenuItem eventKey="35">35 per page</MenuItem>
+										<MenuItem eventKey="50">50 per page</MenuItem>
+										<MenuItem eventKey="100">100 per page</MenuItem>
+									</DropdownButton>
```

Comment:
There's also a nagging styling issue that was here before. My bad !
Now that it's all in one place it's easier to fix :)

This is what I see:
<img width="232" alt="Capture d’écran 2020-02-06 à 12 59 20" src="https://user-images.githubusercontent.com/6179856/73935313-889baf80-48e0-11ea-8feb-ac7ca0d1bd0a.png">

Whatever I did doesn't look good ! :)
I think it is due to some css rules I added willy-nilly. Theres a `.pager li > a, .pager li > span` rule that was probably meant for the 'previous page' and 'next page' buttons only but instead is applied to the dropdown menu elements too.
That can be fixed by making sure we target only `li` elements that are direct children of `.pager`:
`.pager > li > a, .pager > li > span`

Reply:
> I think it is due to some css rules I added willy-nilly. Theres a `.pager li > a, .pager li > span` rule that was probably meant for the 'previous page' and 'next page' buttons only but instead is applied to the dropdown menu elements too.
> That can be fixed by making sure we target only `li` elements that are direct children of `.pager`:
> `.pager > li > a, .pager > li > span`

In stlye.less, there was #searchPage, I renamed it to #pageWithPagination. Is that okay? I think by making it a genric name we can reuse it. Otherwise we would have to add for revisionPage again. And also for editor-page-revision.